cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -100,6 +100,20 @@ private State(ProgramRunStatus runStatus) {
       this.runStatus = runStatus;
     }
 
+    /**
+     * Gives the {@link ProgramController.State} for a given {@link ProgramRunStatus}
+     * @param status : the status
+     * @return the state for the status or null if there is no defined state for the given status
+     */
+    public static State getControllerStateByStatus (ProgramRunStatus status) {
```

Comment:
I think it should be fine because the program completed status is written when the program is completed which will map to STOPPED(ProgramRunStatus.COMPLETED) or ERROR(ProgramRunStatus.FAILED) but if we look at the above case if it give us     STARTING(ProgramRunStatus.RUNNING) it will get mapped to RUNNING when it gets written which is what others maps too. So I think it should be fine.